cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/cube/Cube.java

```diff
@@ -40,10 +42,28 @@
   Collection<TimeSeries> query(CubeQuery query) throws Exception;
 
   /**
+
    * Deletes the data specified by {@link CubeQuery} from all the fact tables.
    * @param query query specifies parameters for deletion. groupByTags and resolution fields are not used.
    * @throws Exception
    */
   //todo: create separate DeleteCubeQuery class
   void delete(CubeQuery query) throws Exception;
+
+  /**
+   * Queries data for next available tags after the given list of tags specified by
```

Comment:
Finds all tag value pairs that are present in aggregated time series that satisfy {@ link CubeExploreQuery}.